src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -56,8 +56,7 @@
 _COMMON_TENANT = 'common'
 
 _AUTH_CTX_FACTORY = lambda authority, cache: adal.AuthenticationContext(authority,
-                                                                        cache=cache,
-                                                                        api_version=None)
+                                                                        cache=cache)
```

Comment:
What error did you see? Regenerating recording should not use anything related with this class, or it is a bug i can take a look.
FYI, we do need to set `api-version` to none, so adal will switch to newer api-version at `1.1+`; otherwise tokens issued for webapi associated resource might be invalid. please refer to the issue azuread/azure-activedirectory-library-for-python#54